src/peerbook/libp2p_peerbook.erl

```diff
@@ -365,7 +376,16 @@ terminate(_Reason, _State) ->
 -spec mk_this_peer(libp2p_peer:peer() | undefined, #state{}) -> {ok, libp2p_peer:peer()} | {error, term()}.
 mk_this_peer(CurrentPeer, State=#state{tid=TID}) ->
     SwarmAddr = libp2p_swarm:pubkey_bin(TID),
-    ListenAddrs = libp2p_config:listen_addrs(TID),
+    Opts = libp2p_swarm:opts(TID),
+    AllowRFC1918 = libp2p_config:get_opt(Opts, [?MODULE, allow_rfc1918], ?DEFAULT_PEERBOOK_ALLOW_RFC1918),
```

Comment:
Maybe move allow to a function and use it her and in the put/3 function usage above? Because the way it's coded now we allow rfc1918 in put but don't mk_peer one?

Reply:
This is intentional because we need to transition through the old nodes publishing 1918 addresses, once we've upgraded the network we can enable rejection of peers with 1918 addresses in them.

Reply:
AH yes, that makes total sense.. Would still be nice not to repeat ourselves though